enhancements/20190919-OVN-Kubernetes.md

```diff
@@ -0,0 +1,114 @@
+---
+title: ovn-kubernetes-ga
+authors:
+- @pecameron
+- @squeed
+- @knobunc
+- @dcbw
+reviewers:
+- TBD
+approvers:
+- TBD
+￼
+creation-date: 2019-09-19
+last-updated: 2019-09-19
+status: implementable
+see-also:
+replaces:
+Superseded-by:
+
+---
+# OVN-Kubernetes
+## Release Signoff Checklist
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+Makes OVN-Kubernetes production-ready for OpenShift’s use-cases.
+
+### Motivation
+OVN centralizes engineering efforts between teams, as well as allowing us
+to iterate much more quickly for advanced networking features.
+
+OVN-Kubernetes allows us to engage an existing upstream community, rather
+than developing an openshift-specific network plugin in isolation.
+
+## Proposal
+OVN-Kubernetes has been in production in non-OpenShift clusters for some
+time. It consists of a central controller, a data store, and individual
+node-level processes.
+
+The central controller has two components: the ovn-kubernetes controller,
+which reconciles Kubernetes objects in to logical OVN objects, and
+ovn-northd, which converts logical objects in to flows.
+
+The data store is a replicated database, with a traditional master-backup
+architecture.
+
+The node-level process (ovn-southd) is responsible for programming the
+flows from ovn-northd in to the vswitch on the local node. It also handles
+some bookkeeping.
+
+#### Controller
+There will be a controller daemonset on every master node. They will
+perform leader election directly via the Kubernetes API. The ovn-kubernetes
+process will, when it achieves leadership, will be responsible for
+starting the ovn-north process.
+
+#### Data store
+We will write a small controller that, based on the result of the leader
+election process, configures the databases (ovn-ndbd and ovn-sbdb) as
+master or backup. The master will be distributed via an endpoint. If a
+node is backup, it will watch for endpoint changes and update replication
+configuration as fit.
+
+#### Node-level
+The node-level components are configured as a daemonset. They run on all
+nodes, including the control-plane. They watch for endpoint changes,
+watch the sbdb for changes, and configure the nodes.
+
+### Test Plan
+OVN-Kubernetes must pass all standard QE tests, as well as the full e2e
+suite. It does not need any special testing, as the GA release does not
+enable any new features.
```

Comment:
I don’t think this is sufficient.  I’d expect to see a full test suite verifying the failover and recovery of the OVN control plane in a number of failure modes that give confidence of this somewhat novel architecture.

That suite would be required to be stable and comprehensive prior to GA